docs/build/guides/transactions/claimable-balances.mdx

```diff
@@ -68,368 +68,623 @@ Each of these accounts can only claim the balance under unique conditions. Accou
+	defer rpcClient.Close()
+
+	fmt.Printf("Looking up balance ID: %s\n", BALANCE_ID)
+
+	// Parse claimable balance ID from hex
+	var claimableBalanceID xdr.ClaimableBalanceId
+	err := xdr.SafeUnmarshalHex(BALANCE_ID, &claimableBalanceID)
+	panicIf(err)
+
+	// Create ledger key for claimable balance
+	ledgerKey := xdr.LedgerKey{
+		Type: xdr.LedgerEntryTypeClaimableBalance,
+		ClaimableBalance: &xdr.LedgerKeyClaimableBalance{
+			BalanceId: claimableBalanceID,
+		},
+	}
+
+	// Convert ledger key to base64 for RPC call
+	ledgerKeyB64, err := xdr.MarshalBase64(ledgerKey)
+	panicIf(err)
+
+	fmt.Println("Fetching from RPC server...")
+
+	// Use GetLedgerEntries method from client
+	ctx := context.Background()
+	resp, err := rpcClient.GetLedgerEntries(ctx, protocol.GetLedgerEntriesRequest{
+		Keys: []string{ledgerKeyB64},
+	})
+	panicIf(err)
+
+	if len(resp.Entries) > 0 {
+		entry := resp.Entries[0]
+		fmt.Println("Found claimable balance")
+
+		// Parse the ledger entry XDR
+		var ledgerEntryData xdr.LedgerEntryData
+		err = xdr.SafeUnmarshalBase64(entry.DataXDR, &ledgerEntryData)
+		panicIf(err)
+
+		claimableBalance := ledgerEntryData.ClaimableBalance
+
+		// Display details
+		fmt.Printf("Amount: %d\n", int64(claimableBalance.Amount))
+		fmt.Printf("Asset: %s\n", claimableBalance.Asset.String())
+		fmt.Printf("Number of claimants: %d\n", len(claimableBalance.Claimants))
+
+		// Show claimants
+		fmt.Println("\nClaimants:")
+		for i, claimant := range claimableBalance.Claimants {
+			address := claimant.V0.Destination.Address()
+			fmt.Printf("   %d. %s\n", i+1, address)
+		}
+	} else {
+		fmt.Println("Claimable balance not found")
+	}
+}
 ```
 
 </CodeExample>
 
-At this point, the `ClaimableBalanceEntry` exists in the ledger, but we’ll need its Balance ID to claim it, which can be done in several ways:
-
-1. The submitter of the entry (Account A in this case) can retrieve the Balance ID before submitting the transaction;
-2. The submitter parses the XDR of the transaction result’s operations; or
-3. Someone queries the list of claimable balances.
-
-Either party could also check the [`/effects`](../../../data/apis/horizon/api-reference/resources/effects/README.mdx) of the transaction or query [`/claimable_balances`](../../../data/apis/horizon/api-reference/resources/claimablebalances/README.mdx) with different filters in Horizon. Note that while (1) may be unavailable in some SDKs as it's just a helper, the other methods are universal.
+With the Claimable Balance ID acquired, either Account B or A can actually submit a claim, depending on which predicate is fulfilled. We’ll assume here that a minute has passed, so Account A just reclaims the balance entry.
 
 <CodeExample>
 
 ```js
-// Method 1: Suppose `txResponse` comes from the transaction submission
-// above on testnet.
-const builder = sdk.TransactionBuilder(
-  txResponse.envelope_xdr,
-  Networks.TESTNET,
-);
-console.log("Balance ID (1):", builder.build().getClaimableBalanceId(0));
```

Comment:
I think it's important to highlight that the claimable balance IDs are predictable, i.e. you don't need to submit the operation to know what it's going to be. Maybe it doesn't belong in this part of the flow any more but I don't think it should be dropped entirely.